src/js/pie.js

```diff
@@ -0,0 +1,200 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    // Draw the pie part of the pie chart. Features include:
+    // 1. able to draw a pie based on the data set and pie options
+    // 2. able to register D3 DOM event listeners
+    // 3. extend the input data set structure to accept an array of objects as long as objects contain a "value" element for drawing the pie slice
+    // 4. able to configure the color of each pie slice
+    // 5. update the pie when the data set changes, including adding or removing slices
+
+    fluid.defaults("gpii.chartAuthoring.pieChart.pie", {
+        gradeNames: ["fluid.viewRelayComponent", "autoInit"],
+        model: {
+            // dataSet accepts:
+            // 1. an array of primitive values, such as numbers;
+            // 2. an array of objects. The "value" element of each object needs to containe the value for drawing each pie slice.
+            // Example: [{id: string, value: number} ... ]
+            dataSet: []
+        },
+        pieOptions: {
+            width: 300,
+            height: 300,
+            colors: null, // An array of colors to fill slices generated for corresponding values from options.dataSet
+            outerRadius: null,
+            innerRadius: null,
+            animationDuration: 750
+        },
+        // The class names to be
+        classNames: {
+            pie: "gpiic-ca-pieChart-pie gpii-ca-pieChart-pie",
+            slice: "gpiic-ca-pieChart-slice gpii-ca-pieChart-slice",
+            text: "gpiic-ca-pieChart-text gpii-ca-pieChart-text"
+        },
+        selectors: {
+            pie: ".gpiic-ca-pieChart-pie",
```

Comment:
You will need to add some validation code which checks that these selectors are indeed of the form "period plus classname" - since this implementation couldn't handle any other form anyway (given that the addressibility of the element is achieved through adding a classname via the d3 "class" directive)